src/chesster.js

```diff
@@ -118,6 +118,169 @@ function leagueDMResponse(patterns, responseName) {
 leagueResponse(['captain guidelines'], 'formatCaptainGuidelinesResponse');
 leagueDMResponse(['captains', 'captain list'], 'formatCaptainsResponse');
 
+/* 
+ * strip components
+ * Splits the text. Adds each word to an object using the keys provided. 
+ *
+ * Each value, component, is made to be lower case.
+ *
+ * If the number of components is fewer than the number of words in the text,
+ * the remaining words will be joined as the final argument.
+ */
+function stripComponents(text, componentKeys){
+    var numComponents = componentKeys.length;
+    var toStrip = _.toLower(text);
+
+    var tokens = _.split(toStrip, " ");
+    var args = _.slice(tokens, 0, componentKeys.length-1);
+    var lastArg = _.slice(tokens, componentKeys.length-1).join(" ");
+
+    if( args.length > numComponents ) 
+        throw new Error("Too few arguments to strip");
+
+    var components = {}; //add each argument to the object, using the keys given
+    _.each(args, function(argument, index){
+        components[componentKeys[index]] = tokens[index];
+    });
+    components[componentKeys[numComponents-1]] = lastArg;
+    
+    return components;
+}
+
+/* availability */
+/* [player <player-name> is] {available, unavailable} for round <round-number> in <league> */
+chesster.hears({
+    middleware: [ 
+        slack.withLeague
+    ],
+    patterns: [
+        'available',
+        'unavailable'
+    ],
+    messageTypes: [
+        'direct_message',
+        'direct_mention'
+    ]
+}, 
+function(bot, message){
+    var heltourOptions = message.league.options.heltour;
+    if(!heltourOptions){
+        winston.error("{} league doesn't have heltour options!?".format(message.league.options.name));
+        return;
+    }
+
+    // I hope I am not over-using the ternary operator here. 
+    // I would like to apply stripComponents to the other functions.
+    // but only if we agree this cleans things up.
+    var components = 
+        (! message.text.includes("player")) ? 
+        stripComponents(
+            message.text, 
+            [ 'available', 'for', 'round', 'roundNumber', 'in','league' ]
```

Comment:
If you go with this style of parsing, I still think you should return the value of the constants that you detected so that you can do things like:
```
var components = tokenize(message.text, ['unavailable|available', 'for', 'round', '{roundNumber}', 'in', '{league}'])
if (components['available']) {
...
} else if (components['unavailable']) {
...
}